pkg/lokifrontend/frontend/v2/frontend.go

```diff
@@ -77,11 +82,17 @@ type Frontend struct {
 
 	schedulerWorkers *frontendSchedulerWorkers
 	requests         *requestsInProgress
+
+	codec queryrangebase.Codec
 }
 
+var _ queryrangebase.Handler = &Frontend{}
+var _ transport.GrpcRoundTripper = &Frontend{}
+
 type frontendRequest struct {
 	queryID      uint64
 	request      *httpgrpc.HTTPRequest
+	queryRequest *queryrange.QueryRequest
```

Comment:
what's this being used for? I tracked it down to being added eventually to the `schedulerRequest` but I can't determine what needs to read it?

Reply:
The scheduler processor will unwrap the requests and apply them. This will avoid the conversion dance from https://github.com/grafana/loki/pull/10858/files#diff-588fc735c566f6b8c69f57b528e2afd9728000fbe80c5004f4646cd6b2044b16R86.